Assets/_BForBoss/_Character/Scripts/CharacterMovementAudio.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using UnityEngine;
+using FMODUnity;
+
+namespace Perigon.Character
+{
+    public class CharacterMovementAudio : MonoBehaviour
+    {
+        [SerializeField] private EventReference _dashAudio;
+        [SerializeField] private EventReference _slideAudio;
+        [SerializeField] private EventReference _jumpAudio;
+
+        [SerializeField] private StudioEventEmitter _runningAudio;
+        [SerializeField] private StudioEventEmitter _wallrunAudio;
+
+        [SerializeField] private float _minSpeed = 0.1f;
+
+        private FirstPersonPlayer _player;
+
+        private MovementSoundState _state;
+        
+        private void Awake()
+        {
+            _player = GetComponent<FirstPersonPlayer>();
+        }
+
+        private void Update()
+        {
+            switch (_state)
+            {
+                case MovementSoundState.NotPlaying:
+                    _state = CheckNotPlaying();
+                    break;
+                case MovementSoundState.Running:
+                    _state = CheckRunning();
+                    break;
+                case MovementSoundState.WallRunning:
+                    _state = CheckWallRunning();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private void OnEnable()
+        {
+            if (_player != null)
+            {
+                _player.Jumped += PlayJumpSound;
+                _player.Dashed += PlayDashSound;
+                _player.Slid += PlaySlideSound;
+            }
+        }
+        
+        private void OnDisable()
+        {
+            if (_player != null)
+            {
+                _player.Jumped -= PlayJumpSound;
+                _player.Dashed -= PlayDashSound;
+                _player.Slid -= PlaySlideSound;
+            }
+        }
+
+        private void PlayJumpSound()
+        {
+            RuntimeManager.PlayOneShot(_jumpAudio, transform.position);
+        }
+        
+        private void PlaySlideSound()
+        {
+            RuntimeManager.PlayOneShot(_slideAudio, transform.position);
+        }
+        
+        private void PlayDashSound()
+        {
+            RuntimeManager.PlayOneShot(_dashAudio, transform.position);
+        }
+
+        private MovementSoundState CheckNotPlaying()
```

Comment:
Can you abstract these `Check___` method, there seems to be copied code within methods. 

Also the method name is misleading as to what is does.  
`Check___` I was assuming it would return some Bool/or void, Where as it's playing audio as well as returning a State.  So the method is doing two in one (playing audio is the side effect)

Reply:
I pulled out the side effect so now it should be a bit more clear what these are doing, and renamed the methods as such